doc/man3/OSSL_HTTP_REQ_CTX.pod

```diff
@@ -0,0 +1,172 @@
+=pod
+
+=head1 NAME
+
+OSSL_HTTP_REQ_CTX,
+OSSL_HTTP_REQ_CTX_new,
+OSSL_HTTP_REQ_CTX_free,
+OSSL_HTTP_REQ_CTX_header,
+OSSL_HTTP_REQ_CTX_add1_header,
+OSSL_HTTP_REQ_CTX_i2d,
+OSSL_HTTP_REQ_CTX_nbio,
+OSSL_HTTP_REQ_CTX_sendreq_d2i,
+OSSL_HTTP_REQ_CTX_get0_mem_bio,
+OSSL_HTTP_REQ_CTX_set_max_response_length
+- HTTP request functions
+
+=head1 SYNOPSIS
+
+ #include <openssl/http.h>
+
+ typedef struct ossl_http_req_ctx_st OSSL_HTTP_REQ_CTX;
+
+ OSSL_HTTP_REQ_CTX *OSSL_HTTP_REQ_CTX_new(BIO *wbio, BIO *rbio,
+                                          int method_GET, int maxline,
+                                          unsigned long max_resp_len,
+                                          int timeout,
+                                          const char *expected_content_type,
+                                          int expect_asn1);
+ void OSSL_HTTP_REQ_CTX_free(OSSL_HTTP_REQ_CTX *rctx);
+
+ int OSSL_HTTP_REQ_CTX_header(OSSL_HTTP_REQ_CTX *rctx,
+                              const char *server,
+                              const char *port, const char *path);
+ int OSSL_HTTP_REQ_CTX_add1_header(OSSL_HTTP_REQ_CTX *rctx,
+                                   const char *name, const char *value);
+
+ int OSSL_HTTP_REQ_CTX_i2d(OSSL_HTTP_REQ_CTX *rctx, const char *content_type,
+                           const ASN1_ITEM *it, ASN1_VALUE *req);
+ int OSSL_HTTP_REQ_CTX_nbio(OSSL_HTTP_REQ_CTX *rctx);
+ ASN1_VALUE *OSSL_HTTP_REQ_CTX_sendreq_d2i(OSSL_HTTP_REQ_CTX *rctx,
+                                           const ASN1_ITEM *it);
+
+ BIO *OSSL_HTTP_REQ_CTX_get0_mem_bio(OSSL_HTTP_REQ_CTX *rctx);
+ void OSSL_HTTP_REQ_CTX_set_max_response_length(OSSL_HTTP_REQ_CTX *rctx,
+                                                unsigned long len);
+
+=head1 DESCRIPTION
+
+B<OSSL_HTTP_REQ_CTX> is a context structure for an HTTP request, used to
+collect all the necessary data to perform that request.
```

Comment:
I suggest adding after this the following paragraph:

This file documents low-level HTTP functions rarely used directly.
High-level HTTP client functions like L<OSSL_HTTP_get(3)> are documented in L<OSSL_HTTP_transfer(3)>.

Reply:
Oh wow, this is the first time I actually look at the "higher level" functions, and that makes me wonder, why do we even have `OSSL_HTTP_REQ_CTX` publically at all?  Does that serve a purpose at all?

Mind you, yes, there are the `OCSP_REQ_CTX` functions, sure, but we don't *have* to make a 1:1 replacement, we only need to replace it with equivalent enough functionality, which the "higher level" OSSL_HTTP functions seem to be...

Reply:
... except of course, since we still need the macros to map `OSCP_REQ_CTX` functions, there must be something to map them to.  I'm quite tempted to propose that `OSSL_HTTP_REQ_CTX` be deprecated as well, which is a bit on the ridiculous side...

Reply:
Anyway, I added those lines nonetheless

Reply:
> why do we even have OSSL_HTTP_REQ_CTX publically at all? Does that serve a purpose at all?

Please see #13873 (keeping in mind that `OSSL_HTTP_REQ_CTX` == `OCSP_REQ_CTX`

Reply:
@levitte pleased to hear that you appreciate the new higher-level API.
As far as that goes, having `OSSL_HTTP_REQ_CTX` public only serves backward compatibility,
but when adding support for session keep-alive (i.e., multiple round trips per session) the context need to be handed out to the caller.